var/spack/repos/builtin/packages/chombo/package.py

```diff
@@ -0,0 +1,130 @@
+       difference and finite-volume methods for the solution of partial
+       differential equations on block-structured adaptively refined
+       logically rectangular (i.e. Cartesian) grids. Note: Chombo download
+       requires human registration process."""
+
+    homepage = "https://anag-repo.lbl.gov/svn/Chombo/release"
+    url      = "https://anag-repo.lbl.gov/svn/Chombo/release/3.2"
+
+    version('3.2', svn='https://anag-repo.lbl.gov/svn/Chombo/release/3.2')
+
+    variant('mpi', default=True, description='Enable MPI parallel support')
+    variant('dims', default=3, description='Number of PDE dimensions [1-6]')
+    variant('hdf5', default=True, description='Enable HDF5 support')
+
+    patch('hdf5-16api.patch', when='@3.2', level=0)
+    patch('Make.defs.local.template.patch', when='@3.2', level=0)
+
+    depends_on('lapack')
+    depends_on('gmake', type='build')
+    depends_on('mpi', when='+mpi')
+    depends_on('hdf5', when='+hdf5')
+    depends_on('hdf5+mpi', when='+mpi+hdf5')
+
+    def edit(self, spec, prefix):
+
+        # Set fortran name mangling in Make.defs
+        defs_file = FileFilter('./lib/mk/Make.defs')
+        defs_file.filter('^\s*#\s*cppcallsfort\s*=\s*',
+                         'cppcallsfort = -DCH_FORT_UNDERSCORE')
+
+        # Set remaining variables in Make.defs.local
+        # Make.defs.local.template.patch ensures lines for USE_TIMER,
+        # USE_LAPACK and lapackincflags are present
+        copyfile('./lib/mk/Make.defs.local.template',
+                 './lib/mk/Make.defs.local')
+
+        defs_file = FileFilter('./lib/mk/Make.defs.local')
+
+        # Unconditional settings
+        defs_file.filter('^\s*#\s*DEBUG\s*=\s*', 'DEBUG = FALSE')
+        defs_file.filter('^\s*#\s*OPT\s*=\s*', 'OPT = TRUE')
+        defs_file.filter('^\s*#\s*PIC\s*=\s*', 'PIC = TRUE')
+        # timer code frequently fails compiles. So disable it.
+        defs_file.filter('^\s*#\s*USE_TIMER\s*=\s*', 'USE_TIMER = FALSE')
+
+        # LAPACK setup
+        defs_file.filter('^\s*#\s*USE_LAPACK\s*=\s*', 'USE_LAPACK = TRUE')
+        defs_file.filter(
+            '^\s*#\s*lapackincflags\s*=\s*',
+            'lapackincflags = -I%s' % spec['lapack'].prefix.include)
+        defs_file.filter(
+            '^\s*#\s*syslibflags\s*=\s*',
+            'syslibflags= -L%s -llapack -lblas' % spec['lapack'].prefix.lib)
+
+        # Compilers and Compiler flags
+        defs_file.filter('^\s*#\s*CXX\s*=\s*', 'CXX = c++')
+        defs_file.filter('^\s*#\s*FC\s*=\s*', 'FC = f90')
+        if '+mpi' in spec:
+            defs_file.filter(
+                '^\s*#\s*MPICXX\s*=\s*',
+                'MPICXX = %s' % self.spec['mpi'].mpicxx)
+
+        # Conditionally determined settings
+        defs_file.filter(
+            '^\s*#\s*MPI\s*=\s*',
+            'MPI = %s' % ('TRUE' if '+mpi' in spec else 'FALSE'))
+        defs_file.filter(
+            '^\s*#\s*DIM\s*=\s*',
+            'DIM = %d' % int(spec.variants['dims'].value))
```

Comment:
> My question is, why is this PR then getting held up

This PR is only 20 hours old.  It is getting expediated review.  The kinds of changes being asked for here are typical of most PRs containing new packages with strange/complex builds.  There are PRs that have been held up many months.

> regarding these minute details when there is already precedent in the existing set of packages? 

Because nobody fixed the previous precedent.  Which is not a good idea to let more stuff we don't want get through.

> Is there some style guide I should have read to have known better ahead of time.

We've talked about making one, but so far no. The fear is it would be so detailed it chases off the average first-committer.  So far, we just have humans review them and bring up the issues as they arise.

> Is there any effort to go back and re-code existing packages to reflect new style/policies in use of Spack?

Yes.